harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -750,19 +661,62 @@ class PyTorchTrial(det.Trial):
     trial_context_class = PyTorchTrialContext
 
     @abstractmethod
+    def __init__(self, trial_context: PyTorchTrialContext) -> None:
+        """
```

Comment:
blocking: we can't add this to the API documentation before we have officially released the API.  As I understand it, this API is not going to be officially released until some time after this PR lands, so this docstring should land at that time.

I *think* that the docstrings for private methods like `_Model` don't get written automatically, so those doc strings would be fine.  But you should double check that.

Reply:
Will move this docstring to `PytorchTrialContext`

Reply:
> blocking: we can't add this to the API documentation before we have officially released the API. As I understand it, this API is not going to be officially released until some time after this PR lands, so this docstring should land at that time.

I think it's fine to add documentation here and call it internal/soft release since all the functionalities are in this PR. I think the underscores can be removed when adding documentation for the new interface since it would be weird to let users use functions starting with underscores in the doc. On the official release, which happens after two rounds of dogfooding, we will deprecate the use of the old interface. We don't need to wait until that time.

Reply:
I think the release plan should be updated per the new API because the old interface proposal is not as invasive as the new one. Here are some thoughts on the constraint and needs we have.

1. Any update on the docstring will show up on the documentation.
2. I don't want to update the docstring until releasing it otherwise the docstring would be inconsistent with the implementation.
3. It's fine to release early before we could tell users we prefer this method. It could say in the documentation. it's a different approach from the current one and it's still experimental.
4. It doesn't make sense to put the interface in `experimental` namespace since it's an improvement on the current interface as opposed to being a risky feature and it's preferred over the older interface. An example that uses a different tactic is the Native project, which is risky and has a large space to improve before we are able to let it replace Trial API.
5. We want to deprecate the old method after internal dogfooding.

The previous plan is to put underscores in the interface functions before releasing it because the training code update might be in different PRs and the new functions would be actually useless.

So I think it's fine to put documentation in the docstring and remove the underscores in this PR since this PR contains the implementation of training for the new interface. I'll remove the underscores and add more docs to the `PytorchTrial` and `PytorchTrialContext` in this PR.

Reply:
To summarize the conversation we had offline, I really don't think it's a good idea to publicly expose an API while we *fully expect* to rename some or possibly all of the functions.